cmd/tool/matrix/matrix.go

```diff
@@ -0,0 +1,250 @@
+		return nil
+	case err != nil:
+		usage(os.Stderr, name, flags)
+		return err
+	}
+	return run(*opts)
+}
+
+type options struct {
+	maxAge            time.Duration
+	buckets           uint
+	timingReportsPath string
+	debug             bool
+}
+
+func setupFlags(name string) (*pflag.FlagSet, *options) {
+	opts := &options{}
+	flags := pflag.NewFlagSet(name, pflag.ContinueOnError)
+	flags.SetInterspersed(false)
+	flags.Usage = func() {
+		usage(os.Stdout, name, flags)
+	}
+	flags.DurationVar(&opts.maxAge, "max-age", 7*24*time.Hour,
+		"Timing reports older than max age will be deleted")
+	flags.UintVar(&opts.buckets, "buckets", 4,
+		"number of parallel buckets to create in the test matrix")
+	flags.StringVar(&opts.timingReportsPath, "dir", "",
+		"path to directory that contains jsonfile timing reports")
+	flags.BoolVar(&opts.debug, "debug", false,
+		"enable debug logging.")
+	return flags, opts
+}
+
+func usage(out io.Writer, name string, flags *pflag.FlagSet) {
+	fmt.Fprintf(out, `Usage:
+    %[1]s [flags]
+
+Flags:
+`, name)
+	flags.SetOutput(out)
+	flags.PrintDefaults()
+}
+
+func run(opts options) error {
+	log.SetLevel(log.InfoLevel)
+	if opts.debug {
+		log.SetLevel(log.DebugLevel)
+	}
+	if opts.buckets < 2 {
+		return fmt.Errorf("--buckets must be atleast 2")
+	}
+	if opts.timingReportsPath == "" {
+		return fmt.Errorf("--dir is required")
+	}
+
+	pkgs, err := readPackages(os.Stdin)
+	if err != nil {
+		return fmt.Errorf("failed to read packages from stdin: %v", err)
+	}
+
+	files, err := readAndPruneTimingReports(opts)
+	if err != nil {
+		return fmt.Errorf("failed to read or delete timing reports: %v", err)
+	}
+	defer closeFiles(files)
+
+	pkgTiming, err := packageTiming(files)
+	if err != nil {
+		return err
+	}
+
+	buckets := bucketPackages(packagePercentile(pkgTiming), pkgs, opts.buckets)
+	return writeBuckets(buckets)
+}
+
+func readPackages(stdin io.Reader) ([]string, error) {
+	var packages []string
+	scan := bufio.NewScanner(stdin)
+	for scan.Scan() {
+		packages = append(packages, scan.Text())
+	}
+	return packages, scan.Err()
+}
+
+func readAndPruneTimingReports(opts options) ([]*os.File, error) {
+	entries, err := os.ReadDir(opts.timingReportsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var files []*os.File
+	for _, entry := range entries {
+		if entry.IsDir() || strings.HasPrefix(entry.Name(), ".") {
+			continue
+		}
+
+		fh, err := os.Open(filepath.Join(opts.timingReportsPath, entry.Name()))
+		if err != nil {
+			return nil, err
+		}
+
+		event, err := parseEvent(fh)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read first event from %v: %v", fh.Name(), err)
+		}
+
+		age := time.Since(event.Time)
+		if age < opts.maxAge {
+			if _, err := fh.Seek(0, io.SeekStart); err != nil {
+				return nil, fmt.Errorf("failed to reset file: %v", err)
+			}
+			files = append(files, fh)
+			continue
+		}
+
+		log.Infof("Removing %v because it is from %v", fh.Name(), event.Time.Format(time.RFC1123))
```

Comment:
Since this will deleted all files at path `--dir=` older than 7 days, it would probably be safer to accept a glob pattern and require a file suffix in the pattern. Something like `./*.log`